src/main/java/net/snowflake/client/core/QueryContextCache.java

```diff
@@ -373,27 +378,17 @@ void logCacheEntries() {
     }
   }
 
-  /**
-   * Note that this schema has been shared to the GS. Changing this may break The query context
-   * contains a list of <id, timestamp, priority, opaque context> tuples. The client will maintain
-   * the lists locally to maintain the latest opaque context for each id, and perform eviction based
-   * on priority (lower numbers have high priorities) to make sure the list size is bounded.
-   */
-  private static final Schema QUERY_CONTEXT_SCHEMA =
-      new Schema(
-          Arrays.asList(
-              createArrowField("id", Types.MinorType.BIGINT.getType(), false),
-              createArrowField("timestamp", Types.MinorType.BIGINT.getType(), false),
-              createArrowField("priority", Types.MinorType.BIGINT.getType(), false),
-              createArrowField("context", Types.MinorType.VARBINARY.getType(), true)));
-
   /** Query context information. */
   private static class QueryContextElement implements Comparable<QueryContextElement> {
     long id; // database id as key. (bigint)
     long readTimestamp; // When the query context read (bigint). Compare for same id.
     long priority; // Priority of the query context (bigint). Compare for different ids.
     byte[] context; // Opaque information (varbinary).
 
+    // Constructor with empty input
+    public QueryContextElement(){
```

Comment:
Why we need constructor without actual values? This may loose integrity. Unless we don't have all values at the same time.

Reply:
In the following function, I first initialize an empty `QueryContextElement`, and then assign the potential value for each field.

```
private static QueryContextElement deserializeQueryContextElement(JsonNode node) throws IOException {
  QueryContextElement entry = new QueryContextElement();
    JsonNode idNode = node.path("id");
    if (idNode.isNumber()) {
        entry.setId(idNode.asLong());
    }

    JsonNode timestampNode = node.path("timestamp");
    if (timestampNode.isNumber()) {
        entry.setReadTimestamp(timestampNode.asLong());
    }

    JsonNode priorityNode = node.path("priority");
    if (priorityNode.isNumber()) {
        entry.setPriority(priorityNode.asLong());
    }

    JsonNode contextNode = node.path("context");
    if (contextNode.isTextual()) {
        byte[] contextBytes = contextNode.asText().getBytes();
        entry.setContext(contextBytes);
    }

    return entry;
}
```